### ISSUE TITLE
Cone specification

### ISSUE THREAD

**Issue by ben-schwen** [oxfordcontrol/clarabel-r#5]
The doc mentions that it is allowed to specify cones in the following way

`cone_spec <- list(z1 = 2L, l1 = 2L, q1 = 2L, zb = 3L, qx = 3L)`

However, then `sanitize_cone_spec` will stop

```r
cone_names <- names(cone_spec)
if (length(intersect(cone_names, c("z", "l", "q", "ep", "p"))) != length(cone_spec)) {
  stop("sanitize_cone_spec: unspecified cone parameters")
}
```

since `intersect` searches for full matches.

**Comment by ben-schwen** [comment]
As a second issue, for the same example once it does not stop it drops all second order cones `q` after the first one.

``` r
cone_spec <- list(z1 = 2L, l1 = 2L, q1 = 2L, zb = 3L, qx = 3L)
clarabel:::sanitize_cone_spec(cone_spec)
#> $z1
#> [1] 2
#> 
#> $l1
#> [1] 2
#> 
#> $q1
#> [1] 3
#> 
#> $zb
#> [1] 3
```

Hence, `qx` is missing

**Comment by bnaras** [comment]
Yes, thank you @ben-schwen .  I discovered that too.  I was trying to emulate `scs` interface, but I think that is not needed.

**Comment by bnaras** [comment]
@ben-schwen I've now released version 0.5.1 which also handles [PSD cones](https://oxfordcontrol.github.io/clarabel-r/articles/clarabel.html#semidefinite-cone-programming). Interface similar to SCS.
Also on this issue, I have clarified in the documentation about the cone documentation. I don't expect people to ever use the internal function `sanitize_cone_spec` except to ensure that the types passed to Rust are correct and the cones are arranged in the order as scs also requires.  Please let me know if this is satisfactory or if I am missing anything.

**Comment by bnaras** [comment]
Closing this as completed.
